File: pkg/services/tmdb.go
package services

import (
	"encoding/json"
	"fmt"
	"net/http"
	"net/url"
	"strconv"

	"neomovies-api/pkg/models"
)

type TMDBService struct {
	accessToken string
	baseURL     string
	client      *http.Client
}

func NewTMDBService(accessToken string) *TMDBService {
	return &TMDBService{
		accessToken: accessToken,
		baseURL:     "https://api.themoviedb.org/3",
		client:      &http.Client{},
	}
}

func (s *TMDBService) makeRequest(endpoint string, target interface{}) error {
	req, err := http.NewRequest("GET", endpoint, nil)
	if err != nil {
		return err
	}

	// Используем Bearer токен вместо API key в query параметрах
	req.Header.Set("Authorization", "Bearer "+s.accessToken)
	req.Header.Set("Accept", "application/json")

	resp, err := s.client.Do(req)
	if err != nil {
		return err
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		return fmt.Errorf("TMDB API error: %d", resp.StatusCode)
	}

	return json.NewDecoder(resp.Body).Decode(target)
}

func (s *TMDBService) SearchMovies(query string, page int, language, region string, year int) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("query", query)
	params.Set("page", strconv.Itoa(page))
	params.Set("include_adult", "false")
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}
	
	if region != "" {
		params.Set("region", region)
	}
	
	if year > 0 {
		params.Set("year", strconv.Itoa(year))
	}

	endpoint := fmt.Sprintf("%s/search/movie?%s", s.baseURL, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) SearchMulti(query string, page int, language string) (*models.MultiSearchResponse, error) {
	params := url.Values{}
	params.Set("query", query)
	params.Set("page", strconv.Itoa(page))
	params.Set("include_adult", "false")

	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/search/multi?%s", s.baseURL, params.Encode())

	var response models.MultiSearchResponse
	err := s.makeRequest(endpoint, &response)
	if err != nil {
		return nil, err
	}

	// Фильтруем результаты: убираем "person", и без названия
	filteredResults := make([]models.MultiSearchResult, 0)
	for _, result := range response.Results {
		if result.MediaType == "person" {
			continue
		}

		hasTitle := false
		if result.MediaType == "movie" && result.Title != "" {
			hasTitle = true
		} else if result.MediaType == "tv" && result.Name != "" {
			hasTitle = true
		}

		if hasTitle {
			filteredResults = append(filteredResults, result)
		}
	}

	response.Results = filteredResults
	response.TotalResults = len(filteredResults)

	return &response, nil
}

func (s *TMDBService) SearchTVShows(query string, page int, language string, firstAirDateYear int) (*models.TMDBTVResponse, error) {
	params := url.Values{}
	params.Set("query", query)
	params.Set("page", strconv.Itoa(page))
	params.Set("include_adult", "false")
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}
	
	if firstAirDateYear > 0 {
		params.Set("first_air_date_year", strconv.Itoa(firstAirDateYear))
	}

	endpoint := fmt.Sprintf("%s/search/tv?%s", s.baseURL, params.Encode())
	
	var response models.TMDBTVResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetMovie(id int, language string) (*models.Movie, error) {
	params := url.Values{}
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/movie/%d?%s", s.baseURL, id, params.Encode())
	
	var movie models.Movie
	err := s.makeRequest(endpoint, &movie)
	return &movie, err
}

func (s *TMDBService) GetTVShow(id int, language string) (*models.TVShow, error) {
	params := url.Values{}
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/tv/%d?%s", s.baseURL, id, params.Encode())
	
	var tvShow models.TVShow
	err := s.makeRequest(endpoint, &tvShow)
	return &tvShow, err
}

func (s *TMDBService) GetGenres(mediaType string, language string) (*models.GenresResponse, error) {
	params := url.Values{}
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/genre/%s/list?%s", s.baseURL, mediaType, params.Encode())
	
	var response models.GenresResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetAllGenres() (*models.GenresResponse, error) {
	// Получаем жанры фильмов
	movieGenres, err := s.GetGenres("movie", "ru-RU")
	if err != nil {
		return nil, err
	}

	// Получаем жанры сериалов
	tvGenres, err := s.GetGenres("tv", "ru-RU")
	if err != nil {
		return nil, err
	}

	// Объединяем жанры, убирая дубликаты
	allGenres := make(map[int]models.Genre)
	
	for _, genre := range movieGenres.Genres {
		allGenres[genre.ID] = genre
	}
	
	for _, genre := range tvGenres.Genres {
		allGenres[genre.ID] = genre
	}

	// Преобразуем обратно в слайс
	var genres []models.Genre
	for _, genre := range allGenres {
		genres = append(genres, genre)
	}

	return &models.GenresResponse{Genres: genres}, nil
}

func (s *TMDBService) GetPopularMovies(page int, language, region string) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}
	
	if region != "" {
		params.Set("region", region)
	}

	endpoint := fmt.Sprintf("%s/movie/popular?%s", s.baseURL, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetTopRatedMovies(page int, language, region string) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}
	
	if region != "" {
		params.Set("region", region)
	}

	endpoint := fmt.Sprintf("%s/movie/top_rated?%s", s.baseURL, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetUpcomingMovies(page int, language, region string) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}
	
	if region != "" {
		params.Set("region", region)
	}

	endpoint := fmt.Sprintf("%s/movie/upcoming?%s", s.baseURL, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetNowPlayingMovies(page int, language, region string) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}
	
	if region != "" {
		params.Set("region", region)
	}

	endpoint := fmt.Sprintf("%s/movie/now_playing?%s", s.baseURL, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetMovieRecommendations(id, page int, language string) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/movie/%d/recommendations?%s", s.baseURL, id, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetSimilarMovies(id, page int, language string) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/movie/%d/similar?%s", s.baseURL, id, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetPopularTVShows(page int, language string) (*models.TMDBTVResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/tv/popular?%s", s.baseURL, params.Encode())
	
	var response models.TMDBTVResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetTopRatedTVShows(page int, language string) (*models.TMDBTVResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/tv/top_rated?%s", s.baseURL, params.Encode())
	
	var response models.TMDBTVResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetOnTheAirTVShows(page int, language string) (*models.TMDBTVResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/tv/on_the_air?%s", s.baseURL, params.Encode())
	
	var response models.TMDBTVResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetAiringTodayTVShows(page int, language string) (*models.TMDBTVResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/tv/airing_today?%s", s.baseURL, params.Encode())
	
	var response models.TMDBTVResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetTVRecommendations(id, page int, language string) (*models.TMDBTVResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/tv/%d/recommendations?%s", s.baseURL, id, params.Encode())
	
	var response models.TMDBTVResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetSimilarTVShows(id, page int, language string) (*models.TMDBTVResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/tv/%d/similar?%s", s.baseURL, id, params.Encode())
	
	var response models.TMDBTVResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

func (s *TMDBService) GetMovieExternalIDs(id int) (*models.ExternalIDs, error) {
	endpoint := fmt.Sprintf("%s/movie/%d/external_ids", s.baseURL, id)
	
	var ids models.ExternalIDs
	err := s.makeRequest(endpoint, &ids)
	return &ids, err
}

func (s *TMDBService) GetTVExternalIDs(id int) (*models.ExternalIDs, error) {
	endpoint := fmt.Sprintf("%s/tv/%d/external_ids", s.baseURL, id)
	
	var ids models.ExternalIDs
	err := s.makeRequest(endpoint, &ids)
	return &ids, err
}

func (s *TMDBService) DiscoverMoviesByGenre(genreID, page int, language string) (*models.TMDBResponse, error) {
	params := url.Values{}
	params.Set("page", strconv.Itoa(page))
	params.Set("with_genres", strconv.Itoa(genreID))
	params.Set("sort_by", "popularity.desc")
	
	if language != "" {
		params.Set("language", language)
	} else {
		params.Set("language", "ru-RU")
	}

	endpoint := fmt.Sprintf("%s/discover/movie?%s", s.baseURL, params.Encode())
	
	var response models.TMDBResponse
	err := s.makeRequest(endpoint, &response)
	return &response, err
}

File: src/services/tmdb.ts
import { config } from "../config"
import { DEFAULT_LANGUAGE } from "../lib/language"
import type {
  TMDBDiscoverParams,
  TMDBEpisode,
  TMDBGenre,
  TMDBMovieDetails,
  TMDBMovie,
  TMDBMultiResult,
  TMDBPageResult,
  TMDBSeasonDetails,
  TMDBTVDetails,
  TMDBTVShow,
  TMDBVideo,
} from "../types/tmdb"
import { TMDBCache } from "./tmdb/cache"
import { buildDefaultParams, filterMovies, filterMulti, filterTV, MIN_VOTE_COUNT } from "./tmdb/filters"

export class TMDBClient {
  private cache = new TMDBCache()
  private baseUrl = config.tmdb.baseUrl
  private headers: Record<string, string>

  constructor() {
    this.headers = {
      Authorization: `Bearer ${config.tmdb.accessToken}`,
      accept: "application/json",
    }
  }

  private generateCacheKey(
    path: string,
    queryParams?: Record<string, string>
  ): string {
    if (!queryParams) return path

    const sortedQuery = Object.entries(queryParams)
      .sort(([a], [b]) => a.localeCompare(b))
      .map(([key, value]) => `${key}=${value}`)
      .join("&")

    return `${path}?${sortedQuery}`
  }

  public async get<T>(
    path: string,
    queryParams?: Record<string, string>
  ): Promise<T> {
    const cacheKey = this.generateCacheKey(path, queryParams)
    const cachedData = this.cache.get<T>(cacheKey)

    if (cachedData) return cachedData

    const url = new URL(`${this.baseUrl}${path}`)
    if (queryParams) {
      Object.entries(queryParams).forEach(([key, value]) => {
        url.searchParams.set(key, value)
      })
    }

    const response = await fetch(url.toString(), { headers: this.headers })
    if (!response.ok) {
      throw new Error(`TMDB API error: ${response.status} ${response.statusText}`)
    }

    const data = (await response.json()) as T
    this.cache.set(cacheKey, data)

    return data
  }

  public imageUrl(path: string | null, size: string = "w500"): string | null {
    if (!path) return null
    return `/image/${size}${path}`
  }

  public imageSizes(
    path: string | null,
    sizes: readonly string[]
  ): Record<string, string | null> {
    const result: Record<string, string | null> = {}

    if (!path) {
      for (const size of sizes) result[size] = null
      return result
    }

    for (const size of sizes) {
      result[size] = `/image/${size}${path}`
    }

    return result
  }

  public async movie(
    id: number,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBMovieDetails> {
    return this.get<TMDBMovieDetails>(`/movie/${id}`, buildDefaultParams(lang))
  }

  public async movieCredits(id: number, lang = DEFAULT_LANGUAGE) {
    return this.get<{ cast: unknown[]; crew: unknown[] }>(
      `/movie/${id}/credits`,
      { language: lang }
    )
  }

  public async movieVideos(
    id: number,
    lang = DEFAULT_LANGUAGE
  ): Promise<{ results: TMDBVideo[] }> {
    return this.get<{ results: TMDBVideo[] }>(
      `/movie/${id}/videos`,
      { language: lang }
    )
  }

  public async tvShow(
    id: number,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBTVDetails> {
    return this.get<TMDBTVDetails>(`/tv/${id}`, buildDefaultParams(lang))
  }

  public async tvSeason(
    id: number,
    season: number,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBSeasonDetails> {
    return this.get<TMDBSeasonDetails>(`/tv/${id}/season/${season}`, {
      language: lang,
    })
  }

  public async tvEpisode(
    id: number,
    season: number,
    episode: number,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBEpisode> {
    return this.get<TMDBEpisode>(
      `/tv/${id}/season/${season}/episode/${episode}`,
      { language: lang }
    )
  }

  public async tvCredits(id: number, lang = DEFAULT_LANGUAGE) {
    return this.get<{ cast: unknown[]; crew: unknown[] }>(
      `/tv/${id}/credits`,
      { language: lang }
    )
  }

  public async tvVideos(
    id: number,
    lang = DEFAULT_LANGUAGE
  ): Promise<{ results: TMDBVideo[] }> {
    return this.get<{ results: TMDBVideo[] }>(
      `/tv/${id}/videos`,
      { language: lang }
    )
  }

  public async tvExternalIds(
    id: number
  ): Promise<{ imdb_id: string | null; tvdb_id: number | null }> {
    return this.get<{ imdb_id: string | null; tvdb_id: number | null }>(
      `/tv/${id}/external_ids`
    )
  }

  public async searchMulti(
    query: string,
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBMultiResult>> {
    const data = await this.get<TMDBPageResult<TMDBMultiResult>>(
      "/search/multi",
      buildDefaultParams(lang, { query, page: String(page) })
    )
    return filterMulti(data)
  }

  public async searchMovie(
    query: string,
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBMovie>> {
    const data = await this.get<TMDBPageResult<TMDBMovie>>(
      "/search/movie",
      buildDefaultParams(lang, { query, page: String(page) })
    )
    return filterMovies(data)
  }

  public async searchTV(
    query: string,
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBTVShow>> {
    const data = await this.get<TMDBPageResult<TMDBTVShow>>(
      "/search/tv",
      buildDefaultParams(lang, { query, page: String(page) })
    )
    return filterTV(data)
  }

  public async discoverMovie(
    params: TMDBDiscoverParams,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBMovie>> {
    const requestParams: Record<string, string> = {
      ...buildDefaultParams(lang),
      "vote_count.gte": String(MIN_VOTE_COUNT),
    }

    for (const [key, value] of Object.entries(params)) {
      if (value !== undefined) requestParams[key] = String(value)
    }

    const data = await this.get<TMDBPageResult<TMDBMovie>>(
      "/discover/movie",
      requestParams
    )
    return filterMovies(data)
  }

  public async discoverTV(
    params: TMDBDiscoverParams,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBTVShow>> {
    const requestParams: Record<string, string> = {
      ...buildDefaultParams(lang),
      "vote_count.gte": String(MIN_VOTE_COUNT),
    }

    for (const [key, value] of Object.entries(params)) {
      if (value !== undefined) requestParams[key] = String(value)
    }

    const data = await this.get<TMDBPageResult<TMDBTVShow>>(
      "/discover/tv",
      requestParams
    )
    return filterTV(data)
  }

  public async popularMovies(
    page = 1,
    lang = DEFAULT_LANGUAGE,
    timeWindow: "day" | "week" = "week"
  ): Promise<TMDBPageResult<TMDBMovie>> {
    const data = await this.get<TMDBPageResult<TMDBMovie>>(
      `/trending/movie/${timeWindow}`,
      buildDefaultParams(lang, { page: String(page) })
    )
    return filterMovies(data)
  }

  public async topRatedMovies(
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBMovie>> {
    return this.discoverMovie(
      {
        page,
        sort_by: "vote_average.desc",
        "vote_count.gte": 1000,
      },
      lang
    )
  }

  public async upcomingMovies(
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBMovie>> {
    const data = await this.get<TMDBPageResult<TMDBMovie>>(
      "/movie/upcoming",
      buildDefaultParams(lang, { page: String(page) })
    )
    return filterMovies(data)
  }

  public async popularTV(
    page = 1,
    lang = DEFAULT_LANGUAGE,
    timeWindow: "day" | "week" = "week"
  ): Promise<TMDBPageResult<TMDBTVShow>> {
    const data = await this.get<TMDBPageResult<TMDBTVShow>>(
      `/trending/tv/${timeWindow}`,
      buildDefaultParams(lang, { page: String(page) })
    )
    return filterTV(data)
  }

  public async topRatedTV(
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBTVShow>> {
    return this.discoverTV(
      {
        page,
        sort_by: "vote_average.desc",
        "vote_count.gte": 500,
      },
      lang
    )
  }

  public async movieGenres(
    lang = DEFAULT_LANGUAGE
  ): Promise<{ genres: TMDBGenre[] }> {
    return this.get<{ genres: TMDBGenre[] }>("/genre/movie/list", {
      language: lang,
    })
  }

  public async tvGenres(
    lang = DEFAULT_LANGUAGE
  ): Promise<{ genres: TMDBGenre[] }> {
    return this.get<{ genres: TMDBGenre[] }>("/genre/tv/list", {
      language: lang,
    })
  }

  public async recommendations(
    type: "movie" | "tv",
    id: number,
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBMovie | TMDBTVShow>> {
    const data = await this.get<TMDBPageResult<TMDBMovie | TMDBTVShow>>(
      `/${type}/${id}/recommendations`,
      buildDefaultParams(lang, { page: String(page) })
    )

    if (type === "movie") {
      return filterMovies(data as TMDBPageResult<TMDBMovie>)
    }
    return filterTV(data as TMDBPageResult<TMDBTVShow>)
  }

  public async similar(
    type: "movie" | "tv",
    id: number,
    page = 1,
    lang = DEFAULT_LANGUAGE
  ): Promise<TMDBPageResult<TMDBMovie | TMDBTVShow>> {
    const data = await this.get<TMDBPageResult<TMDBMovie | TMDBTVShow>>(
      `/${type}/${id}/similar`,
      buildDefaultParams(lang, { page: String(page) })
    )

    if (type === "movie") {
      return filterMovies(data as TMDBPageResult<TMDBMovie>)
    }
    return filterTV(data as TMDBPageResult<TMDBTVShow>)
  }

  public async collection(
    id: number,
    lang = DEFAULT_LANGUAGE
  ): Promise<{
    id: number
    name: string
    overview: string
    poster_path: string | null
    backdrop_path: string | null
    parts: TMDBMovie[]
  }> {
    return this.get(`/collection/${id}`, { language: lang })
  }

  public async movieCertification(id: number): Promise<string | null> {
    try {
      const data = await this.get<{
        results: Array<{
          iso_3166_1: string
          release_dates: Array<{ certification: string }>
        }>
      }>(`/movie/${id}/release_dates`)

      const usRelease = data.results.find((item) => item.iso_3166_1 === "US")
      if (!usRelease) return null

      const cert = usRelease.release_dates.find((item) => item.certification)
      return cert?.certification ?? null
    } catch {
      return null
    }
  }

  public async tvCertification(id: number): Promise<string | null> {
    try {
      const data = await this.get<{
        results: Array<{ rating: string }>
      }>(`/tv/${id}/content_ratings`)

      const usRating = data.results.find((item) => item.rating)
      return usRating?.rating ?? null
    } catch {
      return null
    }
  }
}

export const tmdb = new TMDBClient()
